### ISSUE TITLE
Training and depth related confusion on custom dataset

### ISSUE THREAD

**Issue by poornimajd** [nianticlabs/monodepth2#162]
Hi, great work @mrharicot and @mdfirman .
I trained my model on Indian Driving Dataset. I gave unrectified stereo images as input,and got the following result.
![0000223](https://user-images.githubusercontent.com/31023599/78692562-b66c9700-7917-11ea-9b2c-4bba05a82dbd.jpg)
![0000223_disp](https://user-images.githubusercontent.com/31023599/78692574-bbc9e180-7917-11ea-93b0-c3419abd96ca.jpeg)
May be if I would have given rectified images the results would be better right?
Also in the training command I did not use the option --use_stereo ,but it still by default it takes eigen_zohu and hence the training still defaults to stereo training.
Is this is the case unless we do not write our own custom monodataset loader?

Also since the training is done on stereo images, to calculate the depth , D=f*baseline/disparity  ,with disparity output from this line -
https://github.com/nianticlabs/monodepth2/blob/master/test_simple.py#L137
Is this the correct approach?
 I am sorry if the questions are not appropriate,I am slightly new to this. 
Any suggestion is greatly appreciated.
Thanks

**Comment by mrharicot** [comment]
Hi,
Your questions are totally appropriate!
Great looking results!

What do you mean by it takes eigen_zhou by default? Did you modify the dataloader to take your new input data? 

Now to convert this result to metric depth is going to be trickier as you trained it without rectified pairs.
Do you know the actual baseline in meters and the focal?
How did you train this model exactly? Did you use stereo pairs? Or temporal?

**Comment by poornimajd** [comment]
@mrharicot  thanks for the quick reply.



> What do you mean by it takes eigen_zhou by default? Did you modify the dataloader to take your new input data?

I actually did not modify the dataloader, I arranged my data according to the data-split namely eigen_zhou in kitti format.
 

> Do you know the actual baseline in meters and the focal?

Yes I know the baseline and the focal length


> How did you train this model exactly? Did you use stereo pairs? Or temporal?

I just arranged the stereo images in the format as expected in train_files.txt in splits/eigen_zhou/ as shown below, with the change in name of the images according to the IDD dataset images.
![toshow_mono2](https://user-images.githubusercontent.com/31023599/78707629-7e248300-792e-11ea-9d14-3757b9c86b29.png)

The l and r as usual specify the left and right images
Yes I used stereo pairs,but I used this command for training.
`python3 train.py --model_name mono_model --batch_size 8 --log_dir /media/smart/'My Passport'/mono2_weights/`
As you can see I have not specified --use_stereo parameter,but since I have arranged my data this way , it defaults to stereo training right?
Also 

> Also since the training is done on stereo images, to calculate the depth , D=f*baseline/disparity ,with disparity output from this line -
> https://github.com/nianticlabs/monodepth2/blob/master/test_simple.py#L137

Is this the correct way?
I am not sure if I have messed up something with the training.
Any suggestion greatly appreciated
Thanks

**Comment by msceme** [comment]
You have to use --use_stereo for stereo training.And for stereo training, images should be rectified.

**Comment by poornimajd** [comment]
@msceme  Thank you for the reply.

>  to calculate the depth , D=f*baseline/disparity ,with disparity output from this line -
> https://github.com/nianticlabs/monodepth2/blob/master/test_simple.py#L137
> Is this the correct approach?

Is this the way to calculate the depth ?

**Comment by msceme** [comment]
> @msceme Thank you for the reply.
> 
> > to calculate the depth , D=f*baseline/disparity ,with disparity output from this line -
> > https://github.com/nianticlabs/monodepth2/blob/master/test_simple.py#L137
> > Is this the correct approach?
> 
> Is this the way to calculate the depth ?

If you train on stereo pairs, only then this equation will be applicable.

**Comment by poornimajd** [comment]
Thank you for the quick response!
